src/node_buffer.cc

```diff
@@ -730,12 +730,20 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 
 
 template <typename T, enum Endianness endianness>
-uint32_t WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+void WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  bool noAssert = args[3]->BooleanValue();
+  if (!noAssert) {
```

Comment:
@matthewloring From my distant memory it was indeed faster to coerce `noAssert` in JS. Then you can do a simple `IsTrue()` check, and drop the other two.